# Assignment 3

## importing required packages

In [1]:
import pandas as pd
import numpy as np

## loading the data
### loading the dataset of project here, using functions provided by packages 


In [2]:
data=pd.read_csv(r"C:\Users\Reshu Jain\Downloads\Student_performance_maths.csv")
data.head(10) # data displaying of first ten rows

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
0     GP   F   18       U     GT3       A     4     4   at_home   teacher   
1     GP   F   17       U     GT3       T     1     1   at_home     other   
2     GP   F   15       U     LE3       T     1     1   at_home     other   
3     GP   F   15       U     GT3       T     4     2    health  services   
4     GP   F   16       U     GT3       T     3     3     other     other   
5     GP   M   16       U     LE3       T     4     3  services     other   
6     GP   M   16       U     LE3       T     2     2     other     other   
7     GP   F   17       U     GT3       A     4     4     other   teacher   
8     GP   M   15       U     LE3       A     3     2  services     other   
9     GP   M   15       U     GT3       T     3     4     other     other   

   ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0  ...      4        3      4     1     1      3        6   5   6   6  
1  ...      5        3      3     1     1      3        4   5   5   6  
2  ...      4        3      2     2     3      3       10   7   8  10  
3  ...      3        2      2     1     1      5        2  15  14  15  
4  ...      4        3      2     1     2      5        4   6  10  10  
5  ...      5        4      2     1     2      5       10  15  15  15  
6  ...      4        4      4     1     1      3        0  12  12  11  
7  ...      4        1      4     1     1      1        6   6   5   6  
8  ...      4        2      2     1     1      1        0  16  18  19  
9  ...      5        5      1     1     1      5        0  14  15  15  

[10 rows x 33 columns]

## data description

In [3]:
data.shape # number of rows and columns in a dataset is shown in tuple format

(395, 33)

In [4]:
data.columns# names of all columns of dataset having data type of object

Index(['school', 'sex', 'age', 'address', 'famsize', 'Pstatus', 'Medu', 'Fedu',
       'Mjob', 'Fjob', 'reason', 'guardian', 'traveltime', 'studytime',
       'failures', 'schoolsup', 'famsup', 'paid', 'activities', 'nursery',
       'higher', 'internet', 'romantic', 'famrel', 'freetime', 'goout', 'Dalc',
       'Walc', 'health', 'absences', 'G1', 'G2', 'G3'],
      dtype='object')

In [5]:
data.info() # it gives the information of data . i.e number of columns and the type of values each column contain

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 395 entries, 0 to 394
Data columns (total 33 columns):
school        395 non-null object
sex           395 non-null object
age           395 non-null int64
address       395 non-null object
famsize       395 non-null object
Pstatus       395 non-null object
Medu          395 non-null int64
Fedu          395 non-null int64
Mjob          395 non-null object
Fjob          395 non-null object
reason        395 non-null object
guardian      395 non-null object
traveltime    395 non-null int64
studytime     395 non-null int64
failures      395 non-null int64
schoolsup     395 non-null object
famsup        395 non-null object
paid          395 non-null object
activities    395 non-null object
nursery       395 non-null object
higher        395 non-null object
internet      395 non-null object
romantic      395 non-null object
famrel        395 non-null int64
freetime      395 non-null int64
goout         395 non-null int64
Dalc          395 no

In [6]:
data.size # number of elements in dataset

13035

In [7]:
data.dtypes # datatype of each attribute

school        object
sex           object
age            int64
address       object
famsize       object
Pstatus       object
Medu           int64
Fedu           int64
Mjob          object
Fjob          object
reason        object
guardian      object
traveltime     int64
studytime      int64
failures       int64
schoolsup     object
famsup        object
paid          object
activities    object
nursery       object
higher        object
internet      object
romantic      object
famrel         int64
freetime       int64
goout          int64
Dalc           int64
Walc           int64
health         int64
absences       int64
G1             int64
G2             int64
G3             int64
dtype: object

## data cleaning

### removing column dalc from dataset

In [8]:
data.drop('Dalc',inplace=True,axis=1)
data.head(10)

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
0     GP   F   18       U     GT3       A     4     4   at_home   teacher   
1     GP   F   17       U     GT3       T     1     1   at_home     other   
2     GP   F   15       U     LE3       T     1     1   at_home     other   
3     GP   F   15       U     GT3       T     4     2    health  services   
4     GP   F   16       U     GT3       T     3     3     other     other   
5     GP   M   16       U     LE3       T     4     3  services     other   
6     GP   M   16       U     LE3       T     2     2     other     other   
7     GP   F   17       U     GT3       A     4     4     other   teacher   
8     GP   M   15       U     LE3       A     3     2  services     other   
9     GP   M   15       U     GT3       T     3     4     other     other   

   ... romantic famrel  freetime  goout  Walc health absences  G1  G2  G3  
0  ...       no      4         3      4     1      3        6   5   6   6  
1  ...       no      5         3      3     1      3        4   5   5   6  
2  ...       no      4         3      2     3      3       10   7   8  10  
3  ...      yes      3         2      2     1      5        2  15  14  15  
4  ...       no      4         3      2     2      5        4   6  10  10  
5  ...       no      5         4      2     2      5       10  15  15  15  
6  ...       no      4         4      4     1      3        0  12  12  11  
7  ...       no      4         1      4     1      1        6   6   5   6  
8  ...       no      4         2      2     1      1        0  16  18  19  
9  ...       no      5         5      1     1      5        0  14  15  15  

[10 rows x 32 columns]

### checking missing value for each columns

In [9]:
t=data.isna()
t.head()

school    sex    age  address  famsize  Pstatus   Medu   Fedu   Mjob  \
0   False  False  False    False    False    False  False  False  False   
1   False  False  False    False    False    False  False  False  False   
2   False  False  False    False    False    False  False  False  False   
3   False  False  False    False    False    False  False  False  False   
4   False  False  False    False    False    False  False  False  False   

    Fjob  ...  romantic  famrel  freetime  goout   Walc  health  absences  \
0  False  ...     False   False     False  False  False   False     False   
1  False  ...     False   False     False  False  False   False     False   
2  False  ...     False   False     False  False  False   False     False   
3  False  ...     False   False     False  False  False   False     False   
4  False  ...     False   False     False  False  False   False     False   

      G1     G2     G3  
0  False  False  False  
1  False  False  False  
2  False  False  False  
3  False  False  False  
4  False  False  False  

[5 rows x 32 columns]

In [10]:
t.sum()

school        0
sex           0
age           0
address       0
famsize       0
Pstatus       0
Medu          0
Fedu          0
Mjob          0
Fjob          0
reason        0
guardian      0
traveltime    0
studytime     0
failures      0
schoolsup     0
famsup        0
paid          0
activities    0
nursery       0
higher        0
internet      0
romantic      0
famrel        0
freetime      0
goout         0
Walc          0
health        0
absences      0
G1            0
G2            0
G3            0
dtype: int64

The above output shows that each column has no missing value.

### formatting data
#### change datatype of studytime to float datatype

In [11]:
convert={'studytime':float}
a=data.astype(convert)
print(a.dtypes)

school         object
sex            object
age             int64
address        object
famsize        object
Pstatus        object
Medu            int64
Fedu            int64
Mjob           object
Fjob           object
reason         object
guardian       object
traveltime      int64
studytime     float64
failures        int64
schoolsup      object
famsup         object
paid           object
activities     object
nursery        object
higher         object
internet       object
romantic       object
famrel          int64
freetime        int64
goout           int64
Walc            int64
health          int64
absences        int64
G1              int64
G2              int64
G3              int64
dtype: object


## summary
- ### dataset consists of 395 rows and 33 columns.
- ### total elements in dataset is 13035
- ### 15 int dtypes,17 object dtypes,1 float dtypes.
- ### there is no missing value in rows and columns.